pennylane/spin/spin_hamiltonian.py

```diff
@@ -689,3 +689,67 @@ def kitaev(n_cells, coupling=None, boundary_condition=False):
         hamiltonian += coeff * (opmap[op1](v1) @ opmap[op2](v2))
 
     return hamiltonian.simplify()
+
+
+def spin_hamiltonian(lattice):
+    r"""Generates a spin Hamiltonian for a custom lattice.
+
+    Args:
+        lattice (Lattice): custom lattice defined with ``custom_edges`` and optional ``custom_nodes``
+
+    Raises:
+        ValueError: if the provided Lattice does not have ``custom_edges`` defined with operators
+
+    Returns:
+        ~ops.op_math.Sum: Hamiltonian for the lattice
+
+    **Example**
+
+    .. code-block:: python
+
+        >>> lattice = qml.spin.Lattice(
+            n_cells=[2, 2],
+            vectors=[[1, 0], [0, 1]],
+            positions=[[0, 0], [1, 5]],
+            boundary_condition=False,
+            custom_edges=[[(0, 1), ("XX", 0.5)], [(1, 2), ("YY", 0.6)], [(1, 4), ("ZZ", 0.7)]],
+            custom_nodes=[[0, ("X", 0.5)], [1, ("Y", 0.3)]],
+        )
```

Comment:
This indentation does no seem right too.

Reply:
It's correct... I think the issue is that the rest of the file is using 3 spaces instead of 4 spaces. That's why it seems off.